packages/cozy-harvest-lib/src/components/KonnectorConfiguration/ConfigurationTab/Contracts.jsx

```diff
@@ -38,7 +38,7 @@ const getPrimaryTextPerDoctype = {
 
 const getPrimaryTextDefault = contract => contract.label
 
-const ContractItem = ({ contract }) => {
+const ContractItem = ({ contract, konnector, accountId }) => {
```

Comment:
We have object for two params but only an ID for the third one I find it difficult to remember for a dev POV. Is there a reason to not pass the full `account`?

Reply:
Yes, we want the accountId for the account not to be passed as an object since it needs to be connected from the store directly for the SyncContractSwitch to receive the updated object.

Reply:
Certainement la fin de semaine (mais j'aimerais bien comprendre, donc je persiste ^^"). J'imagine que cet objet `account` il vient d'une requête CozyClient. Et donc si on le passe plus bas, et qu'un composant update, via une requête à la stack, l'objet, on va rafraichir la requête qui a récupéré l'account avec les nouvelles informations et on va donc rafraichir tout son arbre. 

Du coup en partant de là : 
- Soit on n'est pas sur que l'objet vienne du query cozyclient et dans ce cas je comprends 
- Soit on veut que le SyncContractSwitch soit assez autonome pour ne pas dépendre de ses parents (dans ce cas, si t'as une règle pour décider de ça, je suis carrément preneur) 
- Soit on veut pas, pour des raisons de perf, re-rendre tout l'arbre, mais juste assez

- (soit je fais complètement fausse route et je vais me coucher)

Reply:
Pas de problème pour persister. C'est la 1 et donc la 2. C'est un composant super haut dans l'arbre (KonnectorAccounts) qui fetch les accounts et qui les store dans son state ( :( ). Donc j'ai eu une 409 à un moment donné, donc je suis repassé par des ids pour ensuite passer par le store cozy-client, pour que SyncAccountSwitch fonctionne.

Grâce à ta question j'ai pu voir que KonnectorAccounts avait un problème d'architecture, je me le note pour le corriger. Je préfère quand même passer par une étape ou cela fonctionne via un SyncAccountSwitch connecté = je ferais ça dans une autre PR si possible :)

Reply:
Merci pour la réponse ! 😘